### PR TITLE
Sof 1632: Added TV2 action panels

### PR DESCRIPTION
**Pull request by KvelaGorrrrnio**
- TV2 action panel component and TV2 action card component added.
- Camera and video clip actions have specialized rendering.
- Text color in the Rundown view is based on the css variable `--text-color` and is dimmed a bit.
- TV2 action panels can be configured to either wrap the actions to fill up vertical space or use horizontal scrolling to keep a strict layout.
- Showing key bindings are moved to the new component `VirtualKeyboardComponent` in preparation for SOF-1633.

Skærmbillede 2023-10-18 kl  08 20 20

### CODE REVIEW

**Review by andr9528 — APPROVED** [review]

File: src/app/shared/components/tv2-action-panel/tv2-action-panel.component.scss [hunk 1]
```diff
@@ -0,0 +1,58 @@
+:host {
+  position: relative;
+  display: block;
+  border-radius: 8px;
+  margin-bottom: 10px;
+  background-color: #222222;
+  overflow: hidden;
+
+  .title {
+    //color: #cccccc;
```
[the remark this answers is not in the document]
Comment (KvelaGorrrrnio) [reply src/app/shared/components/tv2-action-panel/tv2-action-panel.component.scss]: Nicely spotted :D It has been removed.

File: src/app/keyboard/components/virtual-keyboard/virtual-keyboard.component.html [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+    [class.key--matched]="isKeyBindingMatched(keyBinding)"
+>
+    <span class="key-label">{{ keyBinding.label }}</span>
+    <span class="key-keys">{{ displayKeyBinding(keyBinding) }}</span>
+</div>
+
+<pre class="keystrokes">
+Pressed:keys
+{{ keystrokes }}
+</pre>
```
Comment (LindvedKrvang) [inline src/app/keyboard/components/virtual-keyboard/virtual-keyboard.component.html:13]: Is this still supposed to be here?

File: src/app/producer-shelf/components/producer-shelf/producer-shelf.component.html [hunk 3]
```diff
@@ -1,12 +1,6 @@
-<div
-    *ngFor="let keyBinding of keyBindings" (click)="keyBinding.onMatched()"
-    class="keybinding"
-    [class.keybinding--matched]="isKeyBindingMatched(keyBinding)"
->
-    <span class="keybinding-label">{{ keyBinding.label }}</span>
-    <span class="keybinding-keys">{{ displayKeyBinding(keyBinding) }}</span>
+<div class="action-panels">
+    <sofie-tv2-action-panel *ngIf="cameraActions.length > 0" title="ADLIB PANEL: Camera (Scrollable)" [actions]="cameraActions" (executeAction)="executeAction($event)"></sofie-tv2-action-panel>
```
Comment (LindvedKrvang) [inline src/app/producer-shelf/components/producer-shelf/producer-shelf.component.html:2]: I believe we still want to show the empty panels?
But it seems these are currently more for demo purporses untill we get the actual actions

File: src/app/shared/components/tv2-action-card/tv2-action-card.component.html [hunk 4]
```diff
@@ -0,0 +1 @@
+{{ action.name }}
```
Comment (LindvedKrvang) [inline src/app/shared/components/tv2-action-card/tv2-action-card.component.html:1]: Best kind of markup is no markup :D

**Review by LindvedKrvang — APPROVED** [review]

File: src/app/keyboard/components/virtual-keyboard/virtual-keyboard.component.html (hunk 2 above)
Comment (KvelaGorrrrnio) [reply src/app/keyboard/components/virtual-keyboard/virtual-keyboard.component.html:13]: Yes. I will remove it when the virtual keyboard is implemented :)

File: src/app/producer-shelf/components/producer-shelf/producer-shelf.component.html (hunk 3 above)
Comment (KvelaGorrrrnio) [reply src/app/producer-shelf/components/producer-shelf/producer-shelf.component.html:2]: This is for demo purposes.

**Pull request merged**